src/features/hover.zig

```diff
@@ -345,6 +346,77 @@ fn hoverDefinitionFieldAccess(
     };
 }
 
+fn hoverNumberLiteral(
+    handle: *DocumentStore.Handle,
+    token_index: Ast.TokenIndex,
+    arena: std.mem.Allocator,
+    markup_kind: types.MarkupKind,
+) error{ OutOfMemory, ParseError }!?[]const u8 {
+    const tree = handle.tree;
+    // number literals get tokenized separately from their minus sign
+    const is_negative = tree.tokens.items(.tag)[token_index -| 1] == .minus;
+    const num_slice = tree.tokenSlice(token_index);
+    const parsed = std.zig.parseNumberLiteral(num_slice);
+    // fmt doesn't allow floats to be formatted with binary or hex specifiers
```

Comment:
You would need to `@bitcast` the float to a unsigned integer with the same number of bits like `@bitcast(u32, some_f32)` and then format it.

This is not something that would be very useful since we don't know which float type is being used.

Reply:
Gotcha, that makes sense. Should the comment be removed then?

Reply:
The comment and the if below can be removed. Don't forget to adjust the switch case for .float and .failure to return null instead of unreachable.